src/intern.ts

```diff
@@ -1,219 +1,151 @@
-              } else {
-                if (intern.is_date(qv)) {
-                  if (!(intern.is_date(ev) && intern.eq_dates(qv, ev))) {
-                    continue next_ent
-                  }
-                } else if (qv !== ev) {
-                  continue next_ent
-                }
-              }
-            }
-          }
-          ent = qent.make$(ent)
-          list.push(ent)
-        }
-      }
+    const args = {
+      table_name: ent_table,
+      data: data,
+      id: q
     }
 
-    // Always sort first, this is the 'expected' behaviour.
-    if (null != q && q.sort$) {
-      let sf: any
-      for (sf in q.sort$) {
-        break
-      }
+    const query = Q.upsert(args)
+    return query
+  }
 
-      let sd = q.sort$[sf] < 0 ? -1 : 1
-      list = list.sort(function (a, b) {
-        return sd * (a[sf] < b[sf] ? -1 : a[sf] === b[sf] ? 0 : 1)
-      })
-    }
+  static tablenameUtil(ent: any) {
+    const canon = ent.canon$({ object: true })
 
-    // Skip before limiting.
-    if (null != q && q.skip$ && q.skip$ > 0) {
-      list = list.slice(q.skip$)
-    }
+    return (canon.base ? canon.base + '_' : '') + canon.name
+  }
 
-    // Limited the possibly sorted and skipped list.
-    if (null != q && q.limit$ && q.limit$ >= 0) {
-      list = list.slice(0, q.limit$)
+  static async isNew(ent: any) {
+    const isNew = await intern.findKnex(ent, ent.id)
+
+    if (isNew) {
+      return true
     }
 
-    // Prune fields
-    if (null != q && q.fields$) {
-      for (let i = 0; i < list.length; i++) {
-        let entfields = list[i].fields$()
-        for (let j = 0; j < entfields.length; j++) {
-          if ('id' !== entfields[j] && -1 == q.fields$.indexOf(entfields[j])) {
-            delete list[i][entfields[j]]
-          }
-        }
+    return false
+  }
+
+  static getConfig(spec: any) {
+    let conf
+
+    if ('string' === typeof spec) {
+      const urlM = /^postgres:\/\/((.*?):(.*?)@)?(.*?)(:?(\d+))?\/(.*?)$/.exec(spec)
+
+      if (!urlM)
+      {
+        return null
       }
+      conf = {
+        name: urlM[7],
+        host: urlM[4],
+        username: urlM[2],
+        password: urlM[3],
+        port: urlM[6] ? parseInt(urlM[6], 10) : null,
+      }
+
+    } else {
+      conf = spec
     }
 
-    // Return the resulting list to the caller.
-    done.call(seneca, null, list)
-  }
+    // pg conf properties
+    conf.user = conf.username
+    conf.database = conf.name
 
-  static clean_array(ary: string[]): string[] {
-    return ary.filter((prop: string) => !prop.includes('$'))
-  }
+    conf.host = conf.host || conf.server
+    conf.username = conf.username || conf.user
+    conf.password = conf.password || conf.pass
 
-  static is_object(x: any): boolean {
-    return '[object Object]' === toString.call(x)
+    return conf
   }
 
-  static is_date(x: any): boolean {
-    return '[object Date]' === toString.call(x)
-  }
+  static buildCtx(seneca: any, msg: any, meta: any) {
```

Comment:
remove in this version - we will add tx support separately

Reply:
Ok